tsflex/features/segmenter/strided_rolling.py

```diff
@@ -0,0 +1,371 @@
+        default 'end'.
+    approve_sparsity: bool, optional
+        Bool indicating whether the user acknowledges that there may be sparsity (i.e.,
+        irregularly sampled data), by default False.
+        If False and sparsity is observed, a warning is raised.
+    data_type: Union[np.array, pd.Series], optional
+        The data type of the stroll (either np.array or pd.Series), by default np.array.
+        Note: Make sure to only set this argument to pd.Series when this is really
+        required, since pd.Series strided-rolling is significantly less efficient.
+        For a np.array it is possible to create very efficient views, but there is no
+        such thing as a pd.Series view. Thus, for each stroll, a new series is created.
+
+    Notes
+    -----
+    * This instance withholds a **read-only**-view of the data its values.
+
+    """
+
+    # Create the named tuple
+    _NumpySeriesContainer = namedtuple(
+        "SeriesContainer", ["values", "start_indexes", "end_indexes"]
+    )
+
+    def __init__(
+        self,
+        data: Union[pd.Series, pd.DataFrame, List[Union[pd.Series, pd.DataFrame]]],
+        window: T,
+        stride: T,
+        start_idx: Optional[T] = None,
+        data_type: Optional[Union[np.array, pd.Series]] = np.array,
+        window_idx: Optional[str] = "end",
+        approve_sparsity: Optional[bool] = False,
+    ):
+        self.window = window
+        self.stride = stride
+
+        self.window_idx = window_idx
+        self.approve_sparsity = approve_sparsity
+
+        # TODO: add support for index-data tuple (faster alternative than pd.Series)
+        assert data_type in SUPPORTED_STROLL_TYPES
+        self.data_type = data_type
+
+        # 0. Standardize the input
+        series_list: List[pd.Series] = to_series_list(data)
+        self.series_key: Tuple[str, ...] = tuple([str(s.name) for s in series_list])
+
+        # 1. Determine the start index
+        self.start = start_idx
+        start, self.end = _determine_bounds("inner", series_list)
+        # update self.start if it was not passed
+        self.start = start if self.start is None else start
+
+        # 2. Create a new-index which will be used for DataFrame reconstruction
+        # note: this code can also be placed in the `apply_func` method (if we want to
+        #  make the bound window-idx setting feature specific).
+        # note: the index-name of the first passed series will be used as index-name
+        self.index = self._construct_output_index(series_list[0].index.name)
+
+        # 3. Construct the index ranges and store the series containers
+        np_start_times, np_end_times = self._construct_start_end_times()
+        self.series_containers = self._construct_series_containers(
+            series_list, np_start_times, np_end_times
+        )
+
+        # 4. Check the sparsity assumption
+        if not self.approve_sparsity:
+            last_container = self.series_containers[-1]
```

Comment:
Why only check for the last container (= container that contains the last series of the series list)?
I would loop over all the containers :thinking:

Reply:
good catch!